engine/common/follower/compliance_engine.go

```diff
@@ -216,12 +216,32 @@ func (e *ComplianceEngine) OnFinalizedBlock(block *model.Block) {
 // a blocking manner. It returns the potential processing error when done.
 // This method is intended to be used as a callback by the networking layer,
 // notifying us about fresh proposals directly from the consensus leaders.
+//
+// TODO(BFT, #7620): This function should not return an error. The networking layer's responsibility is fulfilled
+// once it delivers a message to an engine. It does not possess the context required to handle
+// errors that may arise during an engine's processing of the message, as error handling for
+// message processing falls outside the domain of the networking layer.
+//
+// Some of the current error returns signal Byzantine behavior, such as forged or malformed
+// messages. These cases must be logged and routed to a dedicated violation reporting consumer.
 func (e *ComplianceEngine) Process(channel channels.Channel, originID flow.Identifier, message interface{}) error {
 	switch msg := message.(type) {
-	case *messages.UntrustedProposal:
-		e.OnBlockProposal(flow.Slashable[*messages.UntrustedProposal]{
+	case *flow.UntrustedProposal:
+		proposal, err := flow.NewProposal(*msg)
+		if err != nil {
+			e.log.Warn().
```

Comment:
Can we add TODO specifically for replacing this log with call to a protocol violation consumer?